lib/player.js

```diff
@@ -6512,7 +6512,10 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     // Because this.video_ may not be set when the config is built, the default
     // TextDisplay factory must capture a reference to "this".
     config.textDisplayFactory = () => {
-      if (this.videoContainer_) {
+      // On devices where the Fullscreen API is not available we prefer
+      // SimpleTextDisplayer because it works with the Fullscreen API of the
+      // video element itself.
+      if (this.videoContainer_ && document.fullscreenEnabled) {
```

Comment:
I don't think this is the right check.  This could exist and be false.

Maybe it should be `('fullscreenEnabled' in document)`?

Reply:
Oh, wait, I misremembered what that does.  Disregard.